src/main/java/com/example/GymProject/dao/TraineeDao.java

```diff
@@ -3,85 +3,69 @@
 import com.example.GymProject.model.Trainee;
 import jakarta.annotation.PostConstruct;
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
 import org.springframework.stereotype.Repository;
+import org.springframework.util.Assert;
 
-import java.io.BufferedReader;
-import java.io.File;
-import java.io.FileNotFoundException;
-import java.io.FileReader;
+import java.io.*;
 import java.time.LocalDate;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.logging.Logger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @Repository
 public class TraineeDao {
-    private final Logger logger = Logger.getLogger(this.getClass().getName());
+    private final Logger logger = LoggerFactory.getLogger(this.getClass().getName());
     @Value("${traineeFilePath}")
     private String traineeFilePath;
     private final Map<String, Trainee> traineeMap = new HashMap<>();
 
-    public List<Trainee> findAll(){
+    public List<Trainee> findAll() {
         return new ArrayList<>(traineeMap.values());
     }
+    private final ResourceLoader resourceLoader;
 
-    public void create(Trainee trainee){
-        try {
-            if (trainee.getUserId() == null) {
-                logger.warning("Trainee userId is null");
-                throw new IllegalArgumentException("UserId cannot be null");
-            }
-            traineeMap.put(trainee.getUserId(), trainee);
-            logger.info("Inserted New Trainee");
-        } catch (Exception e){
-          logger.severe("Error While Inserting New Value");
-            throw new IllegalArgumentException(e);
-        }
+    public TraineeDao(ResourceLoader resourceLoader) {
+        this.resourceLoader = resourceLoader;
     }
 
-    public Trainee select(String key){
+    public void create(Trainee trainee) {
+
+        Assert.notNull(trainee.getUserId(), "UserId cannot be null");
```

Comment:
What if the whole `trainee` is null? That's the case to check here. At least the first check. After you ensure that trainee is not null, you can check for `trainee.getId()`